src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -617,6 +626,73 @@ async Task<bool> OnPause(SessionId sessionId, JObject args, CancellationToken to
             return true;
         }
 
+        async Task<MethodInfo> LoadSymbolsOnDemand(AssemblyInfo asm, uint method_token, SessionId sessionId, CancellationToken token)
+        {
+            MethodInfo method = null;
+            var context = GetContext(sessionId);
+            ImageDebugHeader header = asm.Image.GetDebugHeader();
+            for (var i = 0; i < header.Entries.Length; i++)
+            {
+                var entry = header.Entries[i];
+                if (entry.Directory.Type == ImageDebugType.CodeView)
+                {
+                    var data = entry.Data;
+
+                    if (data.Length < 24)
+                        return null;
+
+                    var pdbSignature = (data[0]
+                        | (data[1] << 8)
+                        | (data[2] << 16)
+                        | (data[3] << 24));
+
+                    if (pdbSignature != 0x53445352) // "SDSR"
+                        return null;
+
+                    var buffer = new byte[16];
+                    Buffer.BlockCopy(data, 4, buffer, 0, 16);
+
+                    var pdbAge = (data[20]
+                        | (data[21] << 8)
+                        | (data[22] << 16)
+                        | (data[23] << 24));
+
+                    var pdbGuid = new Guid(buffer);
+                    var buffer2 = new byte[(data.Length - 24) - 1];
+                    Buffer.BlockCopy(data, 24, buffer2, 0, (data.Length - 24) - 1);
+                    var pdbName = System.Text.Encoding.UTF8.GetString(buffer2, 0, buffer2.Length);
+                    pdbName = (pdbName.Split(new char[] { '/' }))[pdbName.Split(new char[] { '/' }).Length - 1];
```

Comment:
Is this a filename? Can you use `Path.GetFileName` on it?

Or maybe at least `pdbName = pdbName.Split(new char[] { '/' }) [^1]`
Or maybe `pdbName = pdbName.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries) [^1]`?